app/services/CheckoutService.scala

```diff
@@ -1,24 +1,26 @@
 package services
 
 import com.gu.identity.play.IdMinimalUser
+import com.gu.membership.salesforce.MemberId
 import com.typesafe.scalalogging.LazyLogging
 import model.SubscriptionData
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-class CheckoutService(identityService: IdentityService) extends LazyLogging {
+class CheckoutService(identityService: IdentityService, salesforceService: SalesforceService) extends LazyLogging {
 
   def processSubscription(subscriptionData: SubscriptionData,
-                          idUser: Option[IdMinimalUser]): Future[Either[GuestUserNotCreated, IdMinimalUser]] =
-    idUser.map(user => Future(Right(user)))
+                          idUser: Option[IdMinimalUser]): Future[Option[MemberId]] =
+    idUser.map(user => Future(Some(UserId(user.id))))
       .getOrElse {
         logger.info(s"User does not have an Identity account, creating a Guest one.")
 
-        identityService.registerGuest(subscriptionData.personalData).map { guestFuture =>
-          guestFuture.right.map(id => IdMinimalUser(id, None))
-        }
+        identityService.registerGuest(subscriptionData.personalData)
       }
+      .flatMap(userId =>
+        userId.map(salesforceService.createOrUpdateUser(subscriptionData.personalData, _))
+          .getOrElse(Future(None)))
 }
```

Comment:
Why have we removed the `Either` here too? We should be logging when things go wrong so we can alert on them.